File: src/main/java/dk/itealisten/xmlnormalize/Configuration.java
package dk.itealisten.xmlnormalize;

import java.util.List;

public class Configuration {

    private final List<Node> ignore;
    private final List<Node> sort;

    public Configuration(List<Node> ignore, List<Node> sort) {
        this.ignore = ignore;
        this.sort = sort;
    }

    public List<Node> getIgnore() {
        return ignore;
    }

    public List<Node> getSort() {
        return sort;
    }
}

File: src/main/kotlin/dk/itealisten/xmlnormalize/Configuration.kt
package dk.itealisten.xmlnormalize

class Configuration
constructor(ignoreNodes: Array<Node>, sortNodes: Array<Node>) {

    private val ignore = ArrayList<Node>()
    private val sort = ArrayList<Node>()

    init {
        ignore.addAll(ignoreNodes)
        sort.addAll(sortNodes)
    }

    fun getIgnores(): Array<Node> {
        return ignore.toTypedArray()
    }

    fun getSorts(): Array<Node> {
        return sort.toTypedArray()
    }

    override fun toString(): String {
        return "Configuration(ignore=$ignore, sort=$sort)"
    }


}
